gen-ai/inference/vllm-rayserve-inf2/vllm_asyncllmengine.py

```diff
@@ -0,0 +1,56 @@
+import os
+import json
+
+from fastapi import FastAPI
+from fastapi import Body
+from ray import serve
+from pydantic import BaseModel
+from vllm.engine.arg_utils import AsyncEngineArgs
+from vllm.engine.async_llm_engine import AsyncLLMEngine
+from vllm.sampling_params import SamplingParams
+from vllm.utils import random_uuid
+
+app = FastAPI()
+
+# Define a model for the sampling parameters
+class SamplingParamsModel(BaseModel):
+    temperature: float
+    top_p: float
+
+# Define a model for the request body
+class GenerateRequestModel(BaseModel):
+    prompt: str
+    sampling_params: SamplingParamsModel
+
+@serve.deployment()
+@serve.ingress(app)
+class VLLMDeployment:
+    def __init__(self):
+        engine_args = AsyncEngineArgs(
+            model=os.getenv("MODEL_ID", "TinyLlama/TinyLlama-1.1B-Chat-v1.0"),
```

Comment:
Let's use Llama3 and configure vllm as described here https://gist.github.com/enghwa/dc7c5cf2bfd51751fb626c19facb7f60